trieste/acquisition/function/active_learning.py

```diff
@@ -421,3 +424,107 @@ def __call__(self, x: TensorType) -> TensorType:
+    :cite:`houlsby2011bayesian`.
+    """
+
+    def __init__(self, jitter: float = DEFAULTS.JITTER) -> None:
+        """
+        :param jitter: The size of the jitter to avoid numerical problem caused by the
+                log operation if variance is close to zero.
+        """
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return f"BayesianActiveLearningByDisagreement(jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return bayesian_active_learning_by_disagreement(model, self._jitter)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+class bayesian_active_learning_by_disagreement(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, jitter: float):
+
+        r"""
+        The Bayesian active learning by disagrement acquisition function computes
+        the information gain of the predictive entropy :cite:`houlsby2011bayesian`.
+        the acquisiton function is calculated by:
+
+        .. math::
+            \mathrm{h}\left(\Phi\left(\frac{\mu_{\boldsymbol{x}, \mathcal{D}}}
+            {\sqrt{\sigma_{\boldsymbol{x}, \mathcal{D}}^{2}+1}}\right)\right)
+            -\frac{C \exp \left(-\frac{\mu_{\boldsymbol{x}, \mathcal{D}}^{2}}
+            {2\left(\sigma_{\boldsymbol{w}, \mathcal{D}}^{+C^{2}}\right)}\right)}
+            {\sqrt{\sigma_{\boldsymbol{x}, \mathcal{D}}^{2}+C^{2}}}
+
+        Here :math:`\mathrm{h}(p)` is defined as:
+
+        .. math::
+            \mathrm{h}(p)=-p \log p-(1-p) \log (1-p)
+
+        This acquisition function is intended to use for Binary Gaussian Process Classification
+        model with Bernoulli likelihood. integrating over nuisance parameters is currently not
+        supported (see equation 6 of the paper)
+
+        :param model: The model of the objective function.
+        :param jitter: The size of the jitter to avoid numerical problem caused by the
+                log operation if variance is close to zero.
+        :return: The Bayesian Active Learning By Disagreement acquisition function.
+        """
+        tf.debugging.assert_positive(jitter, message="Jitter must be positive.")
+
+        if type(model) != VariationalGaussianProcess and type(model) != SparseVariational:
```

Comment:
I'm wondering whether it makes sense to require VGP and SVGP, looking at the code below, any classification model that can give you a mean and a variance can be used, no? perhaps we are too restrictive here?

Reply:
I think it is needed, in this version of BALD we need to use the Gaussian model because it uses specific analytical and Taylor approximate of the entropy terms in equation 2 of the paper: see details on [pages 4-5](https://arxiv.org/pdf/1112.5745.pdf).  Not sure if we use a non-gaussian model (for example deep GP) this approximation and analytical form of the entropy terms are still valid or not.  Correct me if I am wrong.

Reply:
ok, I'll check the paper again and get back to you here

forgot to add, here you should use `isinstance` check then this type construction

Reply:
the paper says "Throughout this section we will assume that we are provided with such a Gaussian approximation from one of these methods, though the active learning algorithm does not care which one." where approximation is of the posterior over f - hence it doesn't strictly need to be GPC, but admittedly it's difficult to check that a model is using a Gaussian approx of the posterior...

we are avoiding these kind of direct checks so I think we should remove the checks (and associated tests) and just add a paragraph in the docstring (potentially also notebook) that its been designed for a VGP classifier with Bernoulli likelihood, but other Gaussian approx of the posterior can be used (SVGP for instance, or some other custom model that is not currently supported by Trieste)